### PR TITLE
Upstream release note improvements

### PR DESCRIPTION
**Pull request by jotak**
Small script to grab related components release notes so that they are integrated within the operator release note, which is the one that we advertize more (upstream).

This is more fair as previously, credits to related components contributors were less visible than credits for contributors to the operator itself.

### CODE REVIEW

**Comment by jotak** [comment]
You can check the latest release note to see how it renders: https://github.com/netobserv/network-observability-operator/releases/tag/1.0.2

File: Makefile [hunk 1]
```diff
@@ -367,6 +367,20 @@ catalog-deploy:
 catalog-undeploy:
 	kubectl delete -f ./config/samples/catalog/catalog.yaml
 
+# Grab release notes for related components (to be inserted in operator's release note upstream, cf RELEASE.md)
+.PHONY: related-release-notes
+related-release-notes:
```
Comment (msherif1234) [inline Makefile:372]: pls add ## comment for this new make option so it will show in `make help`

Comment (jotak) [reply Makefile]: I never paid much attention to the `help`, but granted, this is something to care about.
So I just reviewed a little bit and fixed some missing or inaccurate doc in my last commit, also reorganized some sections and moving tools-related to the top.
Here's how it looks like now:

```
$ make help

Usage:
  make <target>

General
  help             Display this help.

Tools
  opm              Download opm locally if necessary.
  controller-gen   Download controller-gen locally if necessary.
  kustomize        Download kustomize locally if necessary.
  envtest          Download envtest-setup locally if necessary.
  crdoc            Download crdoc locally if necessary.
  OPSDK            Download operator-sdk locally if necessary.

Code / files generation
  manifests        Generate WebhookConfiguration, ClusterRole and CustomResourceDefinition objects.
  gencode          Generate code containing DeepCopy, DeepCopyInto, and DeepCopyObject method implementations.
  doc              Generate markdown documentation
  generate-go-conversions  Run all generate-go-conversions
  generate         Run all code/file generators
  clean-generated-conversions  Remove files generated by conversion-gen from the mentioned dirs

Development
  vendors          Refresh vendors directory.
  fmt              Run go fmt against code.
  lint             Run linter (golangci-lint).
  test             Run tests.
  build            Build manager binary.
  image-build      Build operator OCI image.
  image-push       Push OCI image with the manager.

Deployment
  install          Install CRDs into the K8s cluster specified in ~/.kube/config.
  uninstall        Uninstall CRDs from the K8s cluster specified in ~/.kube/config.
  deploy           Deploy controller to the K8s cluster specified in ~/.kube/config.
  undeploy         Undeploy controller from the K8s cluster specified in ~/.kube/config.
  run              Run a controller from your host.

OLM
  bundle-prepare   Generate bundle manifests and metadata, then validate generated files.
  bundle           Generate final bundle files.
  bundle-build     Build the bundle image.
  bundle-push      Push the bundle image.
  catalog-build    Build a catalog image.
  catalog-push     Push a catalog image.
  catalog-deploy   Deploy a catalog image.
  catalog-undeploy  Undeploy a catalog image.

Misc
  related-release-notes  Grab release notes for related components (to be inserted in operator's release note upstream, cf RELEASE.md)

Development helpers
  deploy-loki      Deploy loki.
  undeploy-loki    Undeploy loki.
  undeploy-kafka   Undeploy kafka.
  deploy-grafana   Deploy grafana.
  undeploy-grafana  Undeploy grafana.
  deploy-prometheus  Deploy prometheus.
  undeploy-prometheus  Undeploy prometheus.

Local (Kind)
  install-cert-manager  Install cert manager onto the target kubernetes cluster
  uninstall-cert-manager  Uninstall cert manager from the target kubernetes cluster
  create-kind-cluster  Create kind cluster
  delete-kind-cluster  Delete kind cluster
  local-deploy     Local deploy (kind, loki, grafana, example-cr and sample-workload excluding the operator)
  local-redeploy   Local re-deploy (loki, grafana, example-cr and sample-workload excluding the operator)
  local-undeploy   Local cleanup
  local-run        local-redeploy + run the operator locally
  deploy-kind      Deploy controller to the K8s cluster specified in ~/.kube/config.
  undeploy-kind    Undeploy controller from the K8s cluster specified in ~/.kube/config.

OCP
  ocp-deploy-infra  OCP infra. deploy (only loki and grafana excluding the operator)
  ocp-deploy       OCP deploy (loki, grafana, example-cr and sample-workload excluding the operator)
  ocp-undeploy     OCP cleanup
  ocp-run          OCP-deploy + run the operator locally
  ocp-deploy-operator  run flp from the operator
  undeploy-operator  stop the operator locally
```

Comment (msherif1234) [reply Makefile]: great at some point  we need to orphan local runs targets `ocp-deploy`, `ocp-run` , I don't see the new target u created in this PR inn the help ?

Comment (jotak) [reply Makefile]: it's in `misc`:
```
  related-release-notes  Grab release notes for related components (to be inserted in operator's release note upstream, cf RELEASE.md)
```

**Pull request merged**